src/dotnet-svcutil/lib/src/CodeSerializer.cs

```diff
@@ -0,0 +1,191 @@
+
+namespace Microsoft.Tools.ServiceModel.Svcutil
+{
+    internal class CodeSerializer
+    {
+        static readonly string DefaultFileName = "ServiceReferences";
+        static readonly Encoding ouputEncoding = new System.Text.UTF8Encoding(true);
+
+        readonly CodeDomProvider codeProvider;
+        readonly string outputFilePath;
+
+        internal CodeSerializer(CommandProcessorOptions options, IEnumerable<MetadataSection> inputMetadata)
+        {
+            string extension = GetOutputFileExtension(options);
+            string outputFilename = GetOutputFileName(options, inputMetadata);
+            this.outputFilePath = OutputPathHelper.BuildFilePath(DefaultFileName, options.OutputDir.FullName, outputFilename, extension, CommandProcessorOptions.Switches.OutputFile.Name);
+            this.codeProvider = options.CodeProvider;
+        }
+
+        public string Save(CodeCompileUnit codeCompileUnit)
+        {
+            bool codeGenerated = CompileUnitHasTypes(codeCompileUnit);
+
+            if (!codeGenerated)
+            {
+                throw new ToolRuntimeException(SR.NoCodeWasGenerated);
+            }
+
+            try
+            {
+                return SaveCode(codeCompileUnit);
+            }
+            catch (Exception e)
+            {
+                if (e is ToolRuntimeException || Utils.IsFatalOrUnexpected(e))
+                    throw;
+
+                throw new ToolRuntimeException(SR.GetString(SR.ErrCannotWriteFile), e);
+            }
+        }
+
+        public string SaveCode(CodeCompileUnit codeCompileUnit)
+        {
+            string filePath = null;
+            OutputPathHelper.CreateDirectoryIfNeeded(outputFilePath);
+
+            CodeGeneratorOptions codeGenOptions = new CodeGeneratorOptions();
+            codeGenOptions.BracingStyle = "C";
+
+            using (TextWriter writer = CreateOutputFile())
+            {
+                try
+                {
+                    this.codeProvider.GenerateCodeFromCompileUnit(codeCompileUnit, writer, codeGenOptions);
+                    writer.Flush();
+                }
+                catch (Exception e)
+                {
+                    if (Utils.IsFatalOrUnexpected(e)) throw;
+
+                    try
+                    {
+                        if (File.Exists(this.outputFilePath))
+                        {
+                            File.Delete(this.outputFilePath);
+                        }
+                    }
+                    catch (System.UnauthorizedAccessException)
+                    {
+                    }
+
+                    throw new ToolRuntimeException(SR.GetString(SR.ErrCodegenError), e);
+                }
+                filePath = this.outputFilePath.Contains(" ") ? string.Format(CultureInfo.InvariantCulture, "\"{0}\"", this.outputFilePath) : this.outputFilePath;
+            }
+
+            return filePath;
+        }
+
+        StreamWriter CreateOutputFile()
+        {
```

Comment:
Missing private access modifier. A lot of the statics are too. Could you run the CodeFormatter tool over this code before commiting?